### PR TITLE
Fix capitalization cases for PoshChan

### PR DESCRIPTION
**Pull request by RDIL**
# PR Summary
The jobs name is fully lowercase
Fixes the issue found by @iSazonov 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [x] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by iSazonov** [comment]
The same for "linux"

@SteveL-MSFT Can it be case-insensetive?

**Comment by SteveL-MSFT** [comment]
@PoshChan Please rebuild mAcOS

**Comment by PoshChan** [comment]
@SteveL-MSFT, could not find a matching build target

**Comment by RDIL** [comment]
@SteveL-MSFT typo

**Comment by SteveL-MSFT** [comment]
@RDIL testing case-insensitivity which is what I expected

**Comment by RDIL** [comment]
Oh

**Comment by SteveL-MSFT** [comment]
Intent was to be case-insensitive, but forgot that `ConvertFrom-Json -AsHashTable` results in a case-sensitive hashtable.  I believe I can fix this on the server side.

**Comment by SteveL-MSFT** [comment]
@PoshChan-Staging Please rebuild MACos

**Comment by PoshChan-Staging** [comment]
@SteveL-MSFT, successfully started rebuild of `PowerShell-CI-macOS`

**Comment by SteveL-MSFT** [comment]
@PoshChan please rebuild linux

**Comment by PoshChan** [comment]
@SteveL-MSFT, all requests start with the magic word `Please`.

**Comment by RDIL** [comment]
lol

**Comment by SteveL-MSFT** [comment]
@PoshChan-Staging please rebuild LINux

**Comment by PoshChan-Staging** [comment]
@SteveL-MSFT, successfully started rebuild of `PowerShell-CI-Linux`

**Comment by SteveL-MSFT** [comment]
@PoshChan pLeASE rEbuild wINDOws

**Comment by PoshChan** [comment]
@SteveL-MSFT, successfully started rebuild of `PowerShell-CI-windows`

**Comment by SteveL-MSFT** [comment]
Ok, I believe @PoshChan is fully case-insensitive now :)

So, this PR isn't necessary except for aesthetic reasons.  But then you might as well fix Windows, macOS, and Linux with "correct" casing :)

**Review by SteveL-MSFT — APPROVED** [review]

File: .poshchan/settings.json [hunk 1]
```diff
@@ -3,14 +3,14 @@
     "azdevops": {
         "build_targets": {
             "staging": "PowerShell-CI-static-analysis",
-            "windows": "PowerShell-CI-windows",
+            "windows": "PowerShell-CI-Windows",
             "macos": "PowerShell-CI-macOS",
             "linux": "PowerShell-CI-Linux",
             "all": [
                 "PowerShell-CI-static-analysis",
                 "PowerShell-CI-windows",
```
Comment (SteveL-MSFT) [inline .poshchan/settings.json]:
```suggestion
                "PowerShell-CI-Windows",
```

File: .poshchan/settings.json [hunk 2]
```diff
@@ -3,14 +3,14 @@
...
-                "PowerShell-CI-macOS",
-                "PowerShell-CI-Linux"
+                "PowerShell-CI-macos",
```
Comment (SteveL-MSFT) [inline .poshchan/settings.json]:
```suggestion
                "PowerShell-CI-macOS",
```

File: .poshchan/settings.json [hunk 3]
```diff
@@ -3,14 +3,14 @@
...
+                "PowerShell-CI-linux"
```
Comment (SteveL-MSFT) [inline .poshchan/settings.json]:
```suggestion
                "PowerShell-CI-Linux"
```

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

**Review by SteveL-MSFT — APPROVED** [review]

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**